### ISSUE TITLE
loadmat doesn't work on String arrays

### ISSUE THREAD

**Issue by naterich2** [scipy/scipy#11818]
**Describe the solution you'd like**

I am trying to load a mat file with an array of strings in it.  I have read the documentation and loadmat should work on v6 files, and v7-v7.2.   I have tried saving the mat file from matlab as all of these formats and I have the same issue with each.  The mat file loads with the numerical data, but not the string data.  I've tried to search through the issues and I see that there are issues with implementing classdef's in matlab as well as similar issues to mine come up with the file being loaded with a key 'None' the has a MatlabOpaque object.

**Describe alternatives you've considered**
I've tried saving the string array as a separate array and this produces the same result.  I have also tried all of the ideas in [This stackoverflow question](https://stackoverflow.com/questions/874461/read-mat-files-in-python), and none of them allow me to load the strings.  The h5py loads it in the same way.

At this point I cannot figure out how to load these strings in using any method.

`from scipy.io import loadmat
scipy.__version__`

1.4.1

```
    mat_file = loadmat('data.mat',mat_dtype=True)
    mat_file

    {'__header__': b'MATLAB 5.0 MAT-file, Platform: GLNXA64, Created on: Tue Apr  7       15:07:44 2020',
     '__version__': '1.0',
     '__globals__': [],
     'None': MatlabOpaque([(b'time', b'MCOS', b'string', array([[3707764736],
        [         2],
        [         1],
        [         1],
        [         3],
        [         1]], dtype=uint32))],
              dtype=[('s0', 'O'), ('s1', 'O'), ('s2', 'O'), ('arr', 'O')]),
     '__function_workspace__': array([[ 0,  1, 73, ...,  0,  0,  0]], dtype=uint8)}```

I have also tried setting all the relevant kwargs in the loadmat function with no different results.

**Comment by matthew-brett** [comment]
Yes, I'm afraid we need some more code to deal with Matlab's objects...   But - can you share your .mat file somewhere so we can give it a go?

Can you convert your strings to cell arrays?

**Comment by naterich2** [comment]
I tried converting my strings to cell arrays and that worked!  Hopefully this upload works, let me know if there's more I can do to help!  If nothing else, maybe this workaround could be added to the documentation for loadmat?

I converted it back to the original and saved with `save('test_data.mat', '-v7');`
[test_data.mat.gz](https://github.com/scipy/scipy/files/4451178/test_data.mat.gz)

**Comment by soundarya984** [comment]
Hi, I am facing the same issue. Can you please help. I have changed the data to cell array while saving the .mat file. Still I am unable to decode the array

**Comment by marcosautto** [comment]
Still facing this issue.

**Comment by ahyunSeo** [comment]
Same here

**Comment by Animadversio** [comment]
So what I found is if the string is stored as `char` array in matlab like `str='D:\name'` it will load just fine in python
But if it's the newer `string` like `str="D:\name"` then it won't load properly in python.
So my solution is to load the file in matlab and save all the strings as char aray...

**Comment by Marjaneh-T** [comment]
Same hare.....

**Comment by elijahkin** [comment]
I'm experiencing the same issue with categorical arrays instead of strings. Does anyone know of a solution to this?
